### PR TITLE
correct document picture-in-picture example

### PR DESCRIPTION
**Pull request by joshfarrant**
### Description

Correct mistake in Document Picture-in-Picture API example and improve clarity of docs.

### Motivation

I copy/pasted the current example and it didn't work, so I've tweaked the example a bit to both fix it, and hopefully make it clearer.

### Additional details

N/A

### CODE REVIEW

File: files/en-us/web/api/document_picture-in-picture_api/using/index.md [hunk 1]
```diff
@@ -72,22 +72,24 @@ The `width` and `height` options of `requestWindow()` set the Picture-in-Picture
...
 async function togglePictureInPicture() {
-  // Returns null if no pip window is currently open
-  if (!!window.documentPictureInPicture.window) {
-    // Open a Picture-in-Picture window.
-    const pipWindow = await documentPictureInPicture.requestWindow({
-      width: videoPlayer.clientWidth,
-      height: videoPlayer.clientHeight,
-    });
+  // Early return if there's already a Picture-in-Picture window open
+  if (window.documentPictureInPicture.window) {
```
Comment (skyclouds2001) [inline files/en-us/web/api/document_picture-in-picture_api/using/index.md:76]:
```suggestion
  if (window.documentPictureInPicture.window != null) {
```

personally opinion, favor conditional check over implicit type conversion

File: files/en-us/web/api/document_picture-in-picture_api/using/index.md [hunk 2]
```diff
@@ -72,22 +72,24 @@ The `width` and `height` options of `requestWindow()` set the Picture-in-Picture
...
+    return;
+  }
+
+  // Open a Picture-in-Picture window.
+  const pipWindow = await documentPictureInPicture.requestWindow({
```
Comment (skyclouds2001) [inline files/en-us/web/api/document_picture-in-picture_api/using/index.md:81]:
```suggestion
  const pipWindow = await window.documentPictureInPicture.requestWindow({
```

just be consistent with previous usage of `documentPictureInPicture`

File: files/en-us/web/api/document_picture-in-picture_api/using/index.md (hunk 1 above)
Comment (wbamberg) [reply files/en-us/web/api/document_picture-in-picture_api/using/index.md]: I'm not sure this is our recommended practice actually but have asked around and will get an answer soon.

**Review by wbamberg — COMMENTED** [review]
Thanks for the fix, @joshfarrant !

I don't have a definitive review yet because I want a collective answer from the MDN maintainers on https://github.com/mdn/content/pull/33293#pullrequestreview-2027221099.

But one thing to say is that this guide is intended to mirror the working demo at https://mdn.github.io/dom-examples/document-picture-in-picture/ whose source is at https://github.com/mdn/dom-examples/blob/main/document-picture-in-picture/main.js and it's somewhat important that they should not diverge.

You can see that the dom-examples version is correct: https://github.com/mdn/dom-examples/blob/21320eaacc5f902906f3dc8e3d8d7a75fc1a59aa/document-picture-in-picture/main.js#L19-L23 

```js
	// Returns null if no pip window is currently open
	if (!window.documentPictureInPicture.window) {

		// Open a Picture-in-Picture window.
```

...so we need to:

1. either update the MDN page to match the GitHub example
2. or update the dom-examples version with what's in this PR

I think I favour (2), because I like the early-return pattern (I don't see any guidance either way at https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript). But I'm happy to take care of the GitHub update after this PR is merged.

However I also noticed that [the link in this page to the dom-examples source on GitHub is wrong](https://github.com/mdn/content/blob/3b11785e384572a33ab9202868dadc862a2843d0/files/en-us/web/api/document_picture-in-picture_api/using/index.md?plain=1#L11):

```
> **Note:** You can see the featured demo in action at [Document Picture-in-Picture API Example](https://mdn.github.io/dom-examples/document-picture-in-picture/) (see the full [source code](https://github.com/chrisdavidmills/dom-examples/tree/main/document-picture-in-picture) also).
```

https://github.com/chrisdavidmills/dom-examples/tree/main/document-picture-in-picture
needs to be:
https://github.com/mdn/dom-examples/tree/main/document-picture-in-picture

I would love it if this PR could include that change.

File: files/en-us/web/api/document_picture-in-picture_api/using/index.md (hunk 1 above)
Comment (wbamberg) [reply files/en-us/web/api/document_picture-in-picture_api/using/index.md]: OK, MDN maintainers talked about this and we prefer `if (window.documentPictureInPicture.window)` in this case. Sorry for the confusion.

File: files/en-us/web/api/document_picture-in-picture_api/using/index.md [hunk 3]
```diff
@@ -72,22 +72,24 @@ The `width` and `height` options of `requestWindow()` set the Picture-in-Picture
...
 async function togglePictureInPicture() {
-  // Returns null if no pip window is currently open
-  if (!!window.documentPictureInPicture.window) {
-    // Open a Picture-in-Picture window.
-    const pipWindow = await documentPictureInPicture.requestWindow({
-      width: videoPlayer.clientWidth,
-      height: videoPlayer.clientHeight,
-    });
+  // Early return if there's already a Picture-in-Picture window open
+  if (window.documentPictureInPicture.window != null) {
```
Comment (wbamberg) [inline files/en-us/web/api/document_picture-in-picture_api/using/index.md:76]:
```suggestion
  if (window.documentPictureInPicture.window) {
```

Comment (joshfarrant) [reply files/en-us/web/api/document_picture-in-picture_api/using/index.md]: Nice. I prefer this too 👍

31e582959542a19c23a69353d3abc4dba86b19f4

**Comment by joshfarrant** [comment]
@wbamberg Thanks for the response and detailed comments 💖

I've updated the incorrect link in 05b67eb4359f387197cb353d389a01275204c7a9 and I've opened a PR on the dom-examples repo here https://github.com/mdn/dom-examples/pull/274

Let me know if I've missed anything 🙂

**Review by wbamberg — APPROVED** [review]
Thank you for the fixes, @joshfarrant !

**Pull request merged**